src/syncerd/bitcoin_syncer.rs

```diff
@@ -784,6 +899,41 @@ fn estimate_fee(
     })
 }
 
+fn sweep_polling(
+    state: Arc<Mutex<SyncerState>>,
+    electrum_server: String,
+) -> tokio::task::JoinHandle<()> {
+    tokio::task::spawn(async move {
+        loop {
+            let state_guard = state.lock().await;
+            let sweep_addresses = state_guard.sweep_addresses.clone();
+            drop(state_guard);
+            for (id, sweep_address_task) in sweep_addresses.iter() {
+                if let SweepAddressAddendum::Bitcoin(addendum) = sweep_address_task.addendum.clone()
+                {
+                    let sweep_address_txs = sweep_address(
+                        addendum.private_key,
+                        addendum.address,
+                        addendum.destination_address,
+                        electrum_server.clone(),
+                    )
+                    .unwrap_or_else(|err| {
+                        warn!("error polling sweep address {:?}, retrying", err);
+                        vec![]
+                    });
+                    debug!("sweep address transaction: {:?}", sweep_address_txs);
+                    if !sweep_address_txs.is_empty() {
+                        let mut state_guard = state.lock().await;
+                        state_guard.success_sweep(id, sweep_address_txs).await;
+                        drop(state_guard);
+                    }
```

Comment:
add the else clause to if let Sweep::Bitcoin


this is like above, just copied the msg, change it, please

`else { error!("Aborting sweep address task - unable to decode sweep address addendum"); }`

Reply:
Done in 4ba2040b82d5f8729feae94eecebfb3376f08730